sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py

```diff
@@ -97,110 +93,85 @@ def __init__(self, client, source, **kwargs):
-            )
+            link_properties[EPOCH_SYMBOL] = int(owner_level)
         link_property_timeout_ms = (
-            self._client._config.receive_timeout
-            or self._timeout  # pylint:disable=protected-access
-        ) * 1000
-        self._link_properties[types.AMQPSymbol(TIMEOUT_SYMBOL)] = types.AMQPLong(
-            int(link_property_timeout_ms)
-        )
-        self._handler = None  # type: Optional[ReceiveClient]
+            self._client._config.receive_timeout or self._timeout  # pylint:disable=protected-access
+        ) * self._amqp_transport.IDLE_TIMEOUT_FACTOR
+        link_properties[TIMEOUT_SYMBOL] = int(link_property_timeout_ms)
+        self._link_properties = self._amqp_transport.create_link_properties(link_properties)
+        self._handler: Optional[uamqp_ReceiveClient] = None
         self._track_last_enqueued_event_properties = (
             track_last_enqueued_event_properties
         )
-        self._message_buffer = deque()  # type: Deque[Message]
-        self._last_received_event = None  # type: Optional[EventData]
-        self._receive_start_time = None  # type: Optional[float]
-
-    def _create_handler(self, auth):
-        # type: (JWTTokenAuth) -> None
-        source = Source(self._source)
-        if self._offset is not None:
-            source.set_filter(
-                event_position_selector(self._offset, self._offset_inclusive)
-            )
-        desired_capabilities = None
-        if self._track_last_enqueued_event_properties:
-            symbol_array = [types.AMQPSymbol(RECEIVER_RUNTIME_METRIC_SYMBOL)]
-            desired_capabilities = utils.data_factory(types.AMQPArray(symbol_array))
-
-        properties = create_properties(
-            self._client._config.user_agent  # pylint:disable=protected-access
+        self._message_buffer: Deque[uamqp_Message] = deque()
+        self._last_received_event: Optional[EventData] = None
+        self._receive_start_time: Optional[float]= None
+
+    def _create_handler(self, auth: uamqp_JWTTokenAuth) -> None:
+        source = self._amqp_transport.create_source(
+            self._source,
+            self._offset,
+            event_position_selector(self._offset, self._offset_inclusive)
         )
-        self._handler = ReceiveClient(
-            source,
+        desired_capabilities = [RECEIVER_RUNTIME_METRIC_SYMBOL] if self._track_last_enqueued_event_properties else None
+
+        self._handler = self._amqp_transport.create_receive_client(
+            config=self._client._config,    # pylint:disable=protected-access
+            source=source,
             auth=auth,
-            debug=self._client._config.network_tracing,  # pylint:disable=protected-access
-            prefetch=self._prefetch,
+            network_trace=self._client._config.network_tracing,  # pylint:disable=protected-access
+            link_credit=self._prefetch,
             link_properties=self._link_properties,
-            timeout=self._timeout,
```

Comment:
Why did this one get removed? I'm pretty sure this is not supported by pyamqp yet - but was probably doing something in uamqp?

Reply:
ah I meant to add this in the uamqp_transport.create_receive_client, but didn't add it back in. Since we plan on supporting this in pyamqp, even if we don't support it yet, will add this back in here.